internal/ias/client_test.go

```diff
@@ -0,0 +1,479 @@
+package ias
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/kyma-project/eventing-auth-manager/internal/ias/internal/api"
+	"github.com/kyma-project/eventing-auth-manager/internal/ias/internal/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"testing"
+)
+
+func Test_CreateApplication(t *testing.T) {
+	tests := []struct {
+		name          string
+		givenApiMock  func() *mocks.ClientWithResponsesInterface
+		assertCalls   func(*testing.T, *mocks.ClientWithResponsesInterface)
+		expectedApp   Application
+		expectedError error
+	}{
+		{
+			name: "should create new application when fetching existing applications returns status 200 and no applications",
+			givenApiMock: func() *mocks.ClientWithResponsesInterface {
```

Comment:
It'd be better if you could create a separate generic function for givenApiMock function as well as separate specific functions per use case if generic not possible.